### PR TITLE
sex crimes with tests - new SexCrimes class with ability to check for…

### PR DESCRIPTION
**Pull request by loganballard**
… "romeo and juliet" exceptions, handled by the charge_classifier

Fixes Issue #828

### CODE REVIEW

File: src/backend/expungeservice/models/charge_types/sex_crimes.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+
+@dataclass(eq=False)
+class SexCrimes(Charge):
+    type_name: str = "Sex Crimes"
+    expungement_rules: str = (
+        """
+        Sex Crimes are type-ineligible for expungement, other than a narrow exception for "Romeo and Juliet" cases.
+        According to 137.225(6)(a):
+        Notwithstanding subsection (5) of this section, the provisions of subsection (1)(a) of this section do not apply
```
Comment (wittejm) [inline src/backend/expungeservice/models/charge_types/sex_crimes.py]: I would leave out original statute text here complete text and just have a summarizing note: 
"The required eligibility conditions for this exception are listed in subsection 137.225(6)(a)."

File: src/backend/expungeservice/models/helpers/charge_classifier.py [hunk 2]
```diff
@@ -163,3 +165,8 @@ def _person_felony(statute):
             # In this case the type eligibility needs more analysis. The condition is checked again in the charge object's type eligibility method.
         else:
             return False
+
+    @staticmethod
+    def _sex_crime(statute):
+        if statute in SexCrimes.statutes or statute in SexCrimes.romeo_and_juliet_exceptions:
```
Comment (wittejm) [inline src/backend/expungeservice/models/helpers/charge_classifier.py]: Here's the style we're using elsewhere for this:

```suggestion
        if statute in SexCrimes.statutes + SexCrimes.romeo_and_juliet_exceptions:
```

File: src/backend/expungeservice/models/charge_types/sex_crimes.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    ]
+
+    def _type_eligibility(self):
+        if self.dismissed():
+            return TypeEligibility(EligibilityStatus.ELIGIBLE, reason="Dismissals are eligible under 137.225(1)(b)")
+        elif self.convicted():
+            if self.statute in self.romeo_and_juliet_exceptions:
+                return TypeEligibility(
+                    EligibilityStatus.NEEDS_MORE_ANALYSIS,
+                    reason="Romeo and Juliet exception, needs other elligibility requirements. See 163A.140(1)",
```
Comment (wittejm) [inline src/backend/expungeservice/models/charge_types/sex_crimes.py]:
```suggestion
                    reason="Romeo and Juliet exception, needs other eligibility requirements. See 163A.140(1)",
```

File: src/backend/expungeservice/models/charge_types/sex_crimes.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
+from dataclasses import dataclass
+
+from expungeservice.models.charge import Charge
+from expungeservice.models.expungement_result import TypeEligibility, EligibilityStatus
+
+
+@dataclass(eq=False)
+class SexCrimes(Charge):
+    type_name: str = "Sex Crimes"
```
Comment (wittejm) [inline src/backend/expungeservice/models/charge_types/sex_crimes.py]: Our charges types are all singular, so this should be too (The posted issue was misleading), and that can include the class name and the new annotations in `person_felony.py`

File: src/backend/expungeservice/models/charge_types/sex_crimes.py (hunk 1 above)
Comment (loganballard) [reply src/backend/expungeservice/models/charge_types/sex_crimes.py]: good call, changed

File: src/backend/expungeservice/models/helpers/charge_classifier.py (hunk 2 above)
Comment (loganballard) [reply src/backend/expungeservice/models/helpers/charge_classifier.py]: done

File: src/backend/expungeservice/models/charge_types/sex_crimes.py (hunk 3 above)
Comment (loganballard) [reply src/backend/expungeservice/models/charge_types/sex_crimes.py]: good catch

**Comment by loganballard** [comment]
@wittejm updated based on comments

**Comment by wittejm** [comment]
Yep, lgtm, feel free to merge! Thanks!

**Pull request merged**